### PR TITLE
UI: Status refactoring

### PR DESCRIPTION
**Pull request by mrtracy**
Two commits to remove view code from the stats model.
- Removes all node/store view code from models package - this was used to display some aggregate statistics.
- Converts computations in the node/store models to computed properties.
- General improvement to the way aggregate statistics were being rendered (e.g. no more inline styles, no more tables)

screen shot 2015-08-03 at 4 10 24 pm

### CODE REVIEW

File: ui/ts/models/status.ts [hunk 1]
```diff
@@ -20,87 +21,34 @@ module Models {
...
           });
       });
 
-      constructor() {
-        this.allStatuses = Utils.Computed(this._data.result, (map: StoreStatusMap) => {
-          let keys = Object.keys(map).sort();
-          return keys.map((key: string) => map[key]);
+      private _dataMap: Utils.ReadOnlyProperty<StoreStatusMap> = Utils.Computed(this._data.result, (list: Proto.StoreStatus[]) => {
+        let map: StoreStatusMap = {};
+        _.each(list, (status: Models.Proto.StoreStatus) => {
```
Comment (tamird) [inline ui/ts/models/status.ts]: hm, do you need lodash here? seems like you can use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach?

File: ui/ts/util/format.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // source: util/format.ts
+/// <reference path="../typings/d3/d3.d.ts" />
```
Comment (tamird) [inline ui/ts/util/format.ts]: what made this necessary?

Comment (mrtracy) [inline ui/ts/util/format.ts]: It was necessary for some WIP stuff, but I took it out. Will remove.

File: ui/ts/models/stats.ts [hunk 3]
```diff
@@ -1,63 +0,0 @@
-// source: models/stats.ts
```
Comment (BramGruneir) [inline ui/ts/models/stats.ts]: nice to see this go away :)

File: ui/ts/models/status.ts [hunk 4]
```diff
@@ -110,133 +58,78 @@ module Models {
...
+
+      constructor() {
+        this.allStatuses = this._data.result;
       }
-    }
 
-    export interface NodeStatusResponseSet {
-      d: Proto.NodeStatus[];
-    }
+      public GetTotalStatus(): Proto.Status {
```
Comment (BramGruneir) [inline ui/ts/models/status.ts]: Is this needed?  Why not just make the TotalStatus a public readOnlyProp?

File: ui/ts/models/status.ts [hunk 5]
```diff
@@ -110,133 +58,78 @@ module Models {
...
-        ]);
+        });
+        return status;
+      });
+
+      constructor() {
+        this.allStatuses = this._data.result;
+      }
+
+      public GetTotalStatus(): Proto.Status {
```
Comment (BramGruneir) [inline ui/ts/models/status.ts]: Again, no need for this anymore with the read only prop.

File: ui/ts/models/status.ts [hunk 6]
```diff
@@ -110,133 +58,78 @@ module Models {
...
       });
 
-      constructor() {
-        this.allStatuses = Utils.Computed(this._data.result, (map: NodeStatusMap) => {
-          let keys = Object.keys(map).sort();
-          return keys.map((key: string) => map[key]);
+      private _dataMap: Utils.ReadOnlyProperty<NodeStatusMap> = Utils.Computed(this._data.result, (list: Proto.NodeStatus[]) => {
+        let map: NodeStatusMap = {};
+        _.each(list, (status: Models.Proto.NodeStatus) => {
+          let nodeId: number = status.desc.node_id;
```
Comment (BramGruneir) [inline ui/ts/models/status.ts]: Is there a need for the let?  why not just `map[status.desc.node_id] = status`?

File: ui/ts/models/status.ts [hunk 7]
```diff
@@ -20,87 +21,34 @@ module Models {
...
 
-      constructor() {
-        this.allStatuses = Utils.Computed(this._data.result, (map: StoreStatusMap) => {
-          let keys = Object.keys(map).sort();
-          return keys.map((key: string) => map[key]);
+      private _dataMap: Utils.ReadOnlyProperty<StoreStatusMap> = Utils.Computed(this._data.result, (list: Proto.StoreStatus[]) => {
+        let map: StoreStatusMap = {};
+        _.each(list, (status: Models.Proto.StoreStatus) => {
+          let storeId: number = status.desc.store_id;
+          map[storeId] = status;
```
Comment (BramGruneir) [inline ui/ts/models/status.ts]: why not just `map[status.desc.store_id] = status;` ?

**Comment by BramGruneir** [comment]
LGTM with some minor changes.

File: ui/ts/models/status.ts (hunk 1 above)
Comment (mrtracy) [inline ui/ts/models/status.ts]: I thought lodash _.each would revert to array.forEach when available, but it was actually the other way around; lodash polyfills Array.each with their own each() implementation

File: ui/ts/models/status.ts (hunk 6 above)
Comment (mrtracy) [inline ui/ts/models/status.ts]: I replaced both of these with _.indexBy(), which is lodash's built in method for doing this (moving lists to a map).

File: ui/ts/models/status.ts (hunk 4 above)
Comment (mrtracy) [inline ui/ts/models/status.ts]: Done, it's now a read only prop like `allStatuses`

File: ui/ts/util/format.ts (hunk 2 above)
Comment (mrtracy) [inline ui/ts/util/format.ts]: Hmm, I was actually looking at the wrong reference to `d3.d.ts`. There was indeed a superfluous reference to it in status.ts (also in this code review), but it wasn't added in this commit.

This reference is apparently not needed when compiling the entire project; however, it IS needed if you compile this individual file, which vim's typescript plugin does. This file does reference d3, so it is appropriate to reference those typings.

File: ui/ts/models/stats.ts (hunk 3 above)
Comment (mrtracy) [inline ui/ts/models/stats.ts]: Indeed.

File: ui/ts/models/status.ts (hunk 6 above)
Comment (BramGruneir) [inline ui/ts/models/status.ts]: :+1:

**Pull request merged**